app/services/plan_service/api/plans.rb

```diff
@@ -1,34 +1,99 @@
 module PlanService::API
   PlanStore = PlanService::Store::Plan
 
+  Plan = PlanService::DataTypes::Plan
+
   class Plans
 
+    def initialize(ext_plan_service)
+      @active = Maybe(ext_plan_service)[:active].map {
+        |v| !!v # to bool
+      }.or_else(false)
+    end
+
     def create(community_id:, plan:)
-      Result::Success.new(
-        with_expiration_status(
-          PlanStore.create(community_id: community_id, plan: plan)))
+      if !@active
```

Comment:
We could get rid of all the active checks by having separate implementation for OS use case and external plan service use case. Injector would then choose which one to put in place.